check-versions.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
```

Comment:
Would `check-cmake-with-tag.sh` be slightly more explicit? Also, we are starting to have quite a few scripts at the top level directory of our repo. I wonder if it's worth creating a new `scripts/` directory so that people don't have to scroll too much to get to our `README.md`?

Reply:
It may be, but I'd like to leave this for another PR with some more thought (these are probably `scripts/checks`, there are some in `tests/` that could be moved out, does `start_test_network.sh` get to stay at the top-level for visibility?)